---
title: "Find HTRs"
author: "Lily Peck"
date: "2024-09-17"
output: html_document
---

```{r setup, include=FALSE}
library(tidyverse)
# switch off scientific notation
options(scipen=999)
```


```{r import files}
## first run Fig3_findHTRs.sh using your reference genome and additional set of genomes you want to compare to the reference
## these scripts require the output files from Fig3_findHTRs.sh 
bedG_names <- list.files(path = "./data/10kb_0kbslide", # set the path to your folder with csv files
                        pattern = ".tsv", # select all csv files in the folder
                        full.names = T) # output full file names (with path)
# csv_names <- c("file_1_october.csv", "file_2_november.csv")

bedG_names2 <- data.frame(Type = bedG_names, 
                         id = as.character(1:length(bedG_names))) # id for joining
# import files and tidy up names
bedG <- bedG_names %>% 
  lapply(read_tsv) %>% # read all the files at once
  bind_rows(.id = "id") %>%
  left_join(bedG_names2) %>%
  unite(Chr,Start,End,col = "key", sep = "-") %>%
  mutate(Type = gsub("./data/10kb_0kbslide/","",Type),
         Type = gsub(".minimap2.*$","",Type),
         Type = gsub("GCA.*$","",Type),
         Type = gsub("contigs","",Type)) 

```


```{r link up strain groups into populations / species}
# no Fxyl563 because that is the reference that everything is mapped to
arab <- bedG %>%
  filter(Type == "Fxyl908"|Type == "Fxyl038") %>%
  dplyr::select(!id) %>%
  group_by(key) %>%
  summarise(arab.mean = mean(Similarity))
 
robu <- bedG %>%
  filter(Type == "Fxyl254"|Type == "Fxyl277"|Type == "Fxyl268"|Type == "Fxyl394"|Type == "Fxyl925") %>%
  dplyr::select(!id) %>%
  group_by(key) %>%
  summarise(robu.mean = mean(Similarity))

coff1 <- bedG %>%
  filter(Type == "Fxyl659"|Type == "Fxyl676") %>%
  dplyr::select(!id) %>%
  group_by(key) %>%
  summarise(coff1.mean = mean(Similarity))

coff2 <- bedG %>%
  filter(Type == "Fxyl674"|Type == "Fxyl035"|Type == "Fxyl113") %>%
  dplyr::select(!id) %>%
  group_by(key) %>%
  summarise(coff2.mean = mean(Similarity))

ffc <- bedG %>%
  filter(Type == "Fpro"|Type == "Ffuj"|Type == "Fver"|Type == "Fant"|Type == "Fudu"|Type == "Fphy") %>%
  dplyr::select(!id) %>%
  group_by(key) %>%
  summarise(ffc.mean = mean(Similarity))

foxy <- bedG %>%
  filter(Type == "Fol"|Type == "Foxy109"|Type == "Foxy221"|Type == "Foxy109"|Type == "Foxy248"|Type == "Foxy509"|
         Type == "Foxy541") %>%
  dplyr::select(!id) %>%
  group_by(key) %>%
  summarise(foxy.mean = mean(Similarity))

all <- arab %>%
  left_join(robu, by = "key") %>%
  left_join(coff1, by = "key") %>%
  left_join(coff2, by = "key") %>%
  left_join(ffc, by = "key") %>%
  left_join(foxy, by = "key") %>%
  mutate(key2 = key) %>%
  separate(key2, sep = "-",into = c("contig","start","stop"))  %>%
  mutate(start = as.numeric(start))
```

```{r generate dataframes with putative HTRs}
arab.only <- all %>%
  filter(robu.mean < 0.2 & coff1.mean < 0.2 & coff2.mean < 0.2 & arab.mean > 0.5) %>%
  mutate(type = "arab.only") %>%
  arrange(contig,start)
## have a quick look at data
arab.only %>%
  group_by(contig) %>%
  summarise(length = n()) 
#write_tsv(arab.only,"Summary-arabonly.tsv")

arab.coff1 <- all %>%
  filter(robu.mean < 0.2 & coff1.mean > 0.5 & coff2.mean < 0.2 & arab.mean > 0.7) %>%
  mutate(type = "arab.coff1")
#write_tsv(arab.coff1,"Summary-arabcoff12ref.tsv")

arab.coff2 <- all %>%
  filter(coff1.mean < 0.2 & coff2.mean > 0.5 & arab.mean > 0.7) %>%
  mutate(type = "arab.coff2")
#write_tsv(arab.coff2,"Summary-arabcoff22ref.tsv")

arab.coff1.2 <- all %>%
  filter(robu.mean < 0.2 & coff1.mean > 0.5 & coff2.mean > 0.5 & arab.mean > 0.7)%>%
  mutate(type = "arab.coff1.coff2")
#write_tsv(arab.coff1.2,"Summary-arab.coff1.22ref.tsv")

arab.robu <- all %>%
  filter(robu.mean > 0.5 & coff1.mean < 0.2 & coff2.mean < 0.2 & arab.mean > 0.7) %>%
  mutate(type = "arab.robu")
#write_tsv(arab.robu,"Summary-arabrobu2ref.tsv")

fxyl_pops <- bind_rows(arab.only,arab.coff1,arab.coff2,arab.coff1.2,arab.robu)
fxyl_pops <- fxyl_pops %>%
  dplyr::select(key,contig,start,stop)
```

```{r tidy data for coverage plot}
# format karyotype from long-read assembly
fus_karyotype <- read.table("./data/Fxyl_389563.LRflye.karyo.txt", sep = "\t", header = F, stringsAsFactors = F)
fus_karyotype <- fus_karyotype %>%
  mutate(Start = 1) %>%
  rename(Chr = V1, End = V2) %>%
  relocate(Start, .before = End) %>%
  filter(grepl("contig_", Chr)) 
fus_karyotype$Chr <- gsub("contig_","",fus_karyotype$Chr)
fus_karyotype$Chr <- gsub("_RagTag","",fus_karyotype$Chr)
fus_karyotype$Chr <- as.integer(fus_karyotype$Chr)
# rename contigs as per F. verticillioides chromosomes (mapped in Fig 1A)
fus_karyotype <- fus_karyotype  %>%
  filter(Chr < 16) %>%
  mutate(Chr.new = case_when(Chr == 13 ~ 1.1,
                             Chr == 1 ~ 1.2,
                             Chr == 2 ~ 2,
                             Chr == 4 ~ 3,
                             Chr == 5 ~ 100,
                             Chr == 3 ~ 5,
                             Chr == 12 ~ 6.1,
                             Chr == 15 ~ 6.2,
                             Chr == 14 ~ 6.3,
                             Chr == 8 ~ 6.4,
                             Chr == 6 ~ 200,
                             Chr == 7 ~ 8,
                             Chr == 9 ~ 9,
                             Chr == 10 ~ 10,
                             Chr == 11 ~ 11,
                             T ~ NA
  )) %>%
  arrange(Chr.new) %>%
  mutate(Chr.new = gsub(100,"sc1",Chr.new),
         Chr.new = gsub(200,"sc2",Chr.new))

all2 <- all %>%
  dplyr::select(contig,start,stop,robu.mean,coff1.mean,coff2.mean) %>%
  rename(Chr = contig, Start = start, End = stop, Robusta = robu.mean,
         "Coffea 1" = coff1.mean, "Coffea 2" = coff2.mean) %>%
  mutate(Chr = gsub("contig_","",Chr)) %>%
  mutate(Chr = as.integer(Chr)) %>%
  mutate(Start = as.integer(Start)) %>%
  mutate(End = as.integer(End))  %>%
  filter(Chr < 16) %>%
  mutate(Chr.new = case_when(Chr == 13 ~ 1.1,
                             Chr == 1 ~ 1.2,
                             Chr == 2 ~ 2,
                             Chr == 4 ~ 3,
                             Chr == 5 ~ 100,
                             Chr == 3 ~ 5,
                             Chr == 12 ~ 6.1,
                             Chr == 15 ~ 6.2,
                             Chr == 14 ~ 6.3,
                             Chr == 8 ~ 6.4,
                             Chr == 6 ~ 200,
                             Chr == 7 ~ 8,
                             Chr == 9 ~ 9,
                             Chr == 10 ~ 10,
                             Chr == 11 ~ 11,
                             T ~ NA
  )) %>%
  arrange(Chr.new, Start) %>%
  mutate(Chr.new = gsub(100,"sc1",Chr.new),
         Chr.new = gsub(200,"sc2",Chr.new))
#write_delim(all2,"./data/col2.tsv")
# formatted in excel so the start and end columns are cumulative

## now need to format all the various files for plotting
key <- read_delim("./data/col2.tsv")
key <- key %>%
  unite(Chr:End, col = "key", sep = "-") %>%
  mutate(key = gsub("^","contig_",key)) %>%
  dplyr::select(key,Start.new)
fxyl_pops2 <- fxyl_pops %>%
  left_join(key, by = "key") %>%
  mutate(End.new = (Start.new + 10000))
key <- key %>% dplyr::select(Start.new)
all.new <- bind_cols(all2,key)
row1 <- c(0,0,1,0,0)
fus_karyotype$cum <- cumsum(fus_karyotype$End)
fus_karyotype2 <- rbind(row1,fus_karyotype)
all.new2 <- all.new %>%
  pivot_longer(cols = Robusta:"Coffea 2", names_to = "nams", values_to = "vals") %>%
  drop_na() %>%
  mutate(vals.new = log10(vals)) %>%
  mutate_if(is.numeric, ~ replace_na(., 0) %>% 
                                             replace(., is.infinite(.), -3)) %>%
  dplyr::select(!vals.new) %>%
  mutate(absent = (1-vals)) %>%
  pivot_longer(cols = vals:absent, names_to = "nams2", values_to = "vals2") %>%
  mutate(Colour = case_when(nams == "Robusta" & nams2 == "vals" ~ "RobustaPres",
                            nams == "Coffea 1" & nams2 == "vals" ~ "Coffea1Pres",
                            nams == "Coffea 2" & nams2 == "vals" ~ "Coffea2Pres",
                            .default = "Absent"
                            ))
```

```{r create coverage plot}
pal <- c("white","#F5CDB4", "#F8AFA8","#74A089")

## new plot
all.new2 %>%
  ggplot(aes(x = Start.new, y = vals2, col = Colour)) +
  geom_bar(position="fill", stat="identity") +
  #geom_line(alpha = 1) +
  facet_grid(nams~.,) +
  coord_cartesian(ylim = c(0,1),xlim = c(0,56860843), clip = "off") +
  #coord_cartesian(xlim = c(0,56860843),clip = "off") +
  scale_color_manual(values = pal) +
  annotate(geom="rect", xmin=44831352, xmax=44914251, ymin = -1.55, ymax=-1.65,
           color = "pink", fill="pink") + # contig 11 starship
  annotate(geom = "rect", xmin = fxyl_pops2$Start.new, xmax = fxyl_pops2$End.new, 
           ymin = -1.35, ymax=-1.45, color = "black", fill="black") +
  geom_hline(yintercept = -1.35, col = "lightgrey") +
  geom_hline(yintercept = -1.45, col = "lightgrey") +
  #annotate(geom = "text", x = 0, y = -1.6, label = "HTRs", size = 3) +
  theme_bw() +
  scale_x_continuous(name = "",
                     breaks = fus_karyotype2$cum,
                     labels = fus_karyotype2$Chr.new) +
  # scale_y_continuous(name = "Mapped coverage to arabica reference",
  #                    labels = c(0,25,50,100)) +
  theme(legend.position = "none",
        panel.grid.minor.x = element_blank(),
        panel.grid.major.y = element_blank(),
        panel.grid.minor.y = element_blank(),
        panel.grid.major.x = element_line(colour = "black"),
        axis.title.y = element_text(size = 9),
        axis.text.x = element_blank()) +
  ylab("Mapped coverage to arabica reference") +
  xlab("Fusarium xylarioides chromosomes")
```


```{r plot trees in Fig 3Bi and 3Ci}
library("treeio")
library("ggtree")
## HTR 1 first
tree <- read.tree("./data/htr1.treefile")
## sense check tree first
ggtree(tree) +
  geom_treescale() +
  geom_tiplab() +
  geom_nodelab() +
  xlim(0,10)
## get species groupings for tiplabels
d <- data.frame(label = tree$tip.label)
## table with strain details of BLAST hits
nams <- read_delim("./data/103 documents from contig_13_520000-780000 part 1 - nr Megablast.tsv")
d2 <- d %>%
  rename(Name = label) %>%
  left_join(nams, by = "Name") %>%
  mutate(Label = Description) %>%
  mutate(Label = gsub(" ","_",Label)) %>%
  mutate(Label = gsub('^([^_]+_[^_]+).*', '\\1', Label)) %>%
  mutate(Label = case_when(Label == "Fusarium_verticillioides" ~ "FFC", 
                           Label == "Fusarium_oxysporum" ~ "FOSC",
                           Label == "Fusarium_fujikuroi" ~ "FFC",
                           Label == "Fusarium_xylarioides" ~ "Arabica",
                           grepl("Foxy", Name) ~ "FOSC",
                           Label == "Fusarium_odoratissimum" ~ "FOSC",
                           grepl("Fusarium redolens", Description) ~ "Other",
                           grepl("CM", Name) ~ "FOSC",
                           T ~ "Outgroup"))
pal <- c("#9C964A", "#F98400","grey")

tree %>%
  ggtree() %<+% d2 +
  geom_tiplab(offset = 0.1, size = 3) +
  geom_nodelab(hjust = 1.5, size = 3) +
  geom_tippoint(aes(x = x+.01, color=Label), alpha=0.6,size=5, na.rm = TRUE) +
  geom_treescale() +
  labs(title = "contig 13") +
  scale_color_manual(values = pal) +
  xlim(0,3) +
  theme(legend.position = "bottom",
        legend.title = element_blank())
#ggsave("htr1_tree.svg")

## Then HTR 2
tree <- read.tree("./data/htr2.treefile")
## sense check tree first
ggtree(tree) +
  geom_treescale() +
  geom_tiplab() +
  geom_nodelab() +
  xlim(0,3)
## get species groupings for tiplabels
d <- data.frame(label = tree$tip.label)
## table with strain details of BLAST hits
nams <- read_delim("./data/201 documents from contig_3_RagTag_1310000-1350000 - nr Megablast.tsv", delim = "\t")
d2 <- d %>%
  rename(Name = label) %>%
  left_join(nams, by = "Name") %>%
  mutate(Label = Description) %>%
  mutate(Label = gsub(" ","_",Label)) %>%
  mutate(Label = gsub('^([^_]+_[^_]+).*', '\\1', Label)) %>%
  mutate(Label = case_when(Label == "Fusarium_verticillioides" ~ "FFC", 
                           Label == "Fusarium_oxysporum" ~ "FOSC",
                           Label == "Fusarium_fujikuroi" ~ "FFC",
                           Label == "Fusarium_mangiferae" ~ "FFC",
                           Label == "Fusarium_proliferatum" ~ "FFC",
                           Name == "Fxyl676_contig-3" ~ "Coffea1",
                           Name == "Fxyl659_contig-3" ~ "Coffea1",
                           Name == "Fxyl674_contig-3" ~ "Coffea2",
                           Name == "Fxyl113_contig-3" ~ "Coffea2",
                           Name == "Fxyl035_contig-3" ~ "Coffea2",
                           Name == "Fxyl038_contig-3" ~ "Arabica",
                           Name == "Fxyl563_contig-3" ~ "Arabica",
                           Name == "Fxyl908_contig-3" ~ "Arabica",
                           T ~ "Outgroup"))
pal <- c("#9C964A", "#F5CDB4", "#F8AFA8", "#FDDDA0","#F98400","lightgrey")

tree %>%
  ggtree() %<+% d2 +
  geom_tiplab(offset = 0.07, size = 3) +
  geom_nodelab(hjust = 1.5, size = 3) +
  geom_treescale() +
  geom_tippoint(aes(x = x+.01, color=Label), alpha=0.6,size=5, na.rm = TRUE) +
  labs(title = "Contig 3") +
  scale_color_manual(values = pal) +
  xlim(0,3) +
  theme(legend.position = "bottom",
        legend.title = element_blank())
#ggsave("htr2_tree.svg")
```

```{r coverage plots in Fig 3Bii and 3Cii}
htr <- read_delim("./Data/HTRs-loci-windows.txt")
## HTR 1
## The closest node in Fig 3Bi belongs to GCA_00014955 which is F. oxysporum f. sp. lycopersici
foxy <- bedG %>%
  filter(Type == "Fol") %>%
  mutate(key = gsub("contig_","",key)) %>%
  rename(GCA_000149955 = Similarity) %>%
  dplyr::select(key:GCA_000149955)
plot.tab <- all.new %>%
  unite(Chr:End,col = "key", sep = "-") %>%
  left_join(foxy, by = "key") %>%
  relocate(GCA_000149955, .before = Chr.new) %>%
  pivot_longer(cols = Robusta:GCA_000149955, names_to = "nams", values_to = "vals") %>%
  drop_na() %>%
  mutate(vals.new = log10(vals)) %>%
  mutate_if(is.numeric, ~ replace_na(., 0) %>% 
              replace(., is.infinite(.), -3))
pal <- c("#F5CDB4", "#F8AFA8","#F98400","#74A089")
plot.tab %>%
  ggplot(aes(x = Start.new, y = vals, col = nams)) +
  geom_line(alpha = 1, size = 1) +
  coord_cartesian(xlim = c(430000,800000),ylim=c(0,1), clip = "off") +
  annotate("segment", x = 530000, xend = 730000, y = 1.1, yend = 1.1,colour = "blue") +
  xlim(430000,800000) +  
  scale_color_manual(values = pal) +
  theme_bw() +
  scale_y_continuous(name = "Mapped coverage to arabica reference") +
  theme(legend.position = "bottom",
        legend.title = element_blank(),
        panel.grid.minor.x = element_blank(),
        panel.grid.major.y = element_blank(),
        panel.grid.minor.y = element_blank(),
        panel.grid.major.x = element_blank(),
        axis.title.y = element_blank(),
        axis.title.x = element_blank()) 

## HTR 2
## The closest node in Fig 3Ci belongs to GCA_030345115 which is F. oxysporum, so need to rerun XX.sh to map this genome to Fxyl reference
foxy <- read_delim("./Data/GCA_0303451151_ASM3034511v1_genomic.minimap2_Fxyl_389563.LR.cov.binned.tsv", col_names = F)
foxy <- foxy %>%
  mutate(X1 = gsub("contig_","",X1)) %>%
  unite(X1,X2,X3,col = "key", sep = "-") %>%
  rename(GCA_030345115 = X4)

plot.tab <- all.new %>%
  unite(Chr:End,col = "key", sep = "-") %>%
  left_join(foxy, by = "key") %>%
  relocate(GCA_030345115, .before = Chr.new)%>%
  pivot_longer(cols = Robusta:GCA_030345115, names_to = "nams", values_to = "vals") %>%
  drop_na() %>%
  mutate(vals.new = log10(vals)) %>%
  mutate_if(is.numeric, ~ replace_na(., 0) %>% 
              replace(., is.infinite(.), -3))

pal <- c("#F5CDB4", "#F8AFA8","#F98400","#74A089")
plot.tab %>%
  ggplot(aes(x = Start.new, y = vals, col = nams)) +
  geom_line(alpha = 1, size = 1) +
  coord_cartesian(xlim = c(21600000,21700000),ylim=c(0,1), clip = "off") +
  annotate("segment", x = 21650000, xend = 21670000, y = 1.1, yend = 1.1,colour = "blue") +
  xlim(21600000,21700000) +
  scale_color_manual(values = pal) +
  theme_bw() +
  scale_y_continuous(name = "Mapped coverage to arabica reference") +
  theme(legend.position = "bottom",
        legend.title = element_blank(),
        panel.grid.minor.x = element_blank(),
        panel.grid.major.y = element_blank(),
        panel.grid.minor.y = element_blank(),
        panel.grid.major.x = element_blank(),
        axis.title.y = element_blank(),
        axis.title.x = element_blank()) 

```


```{r HTR 3, Fig S6A}
## first tree
tree <- read.tree("./Data/htr3.treefile")
## table with strain details of BLAST hits
nams <- read_delim("./Data/103 documents from contig_7_4370000-4400000 - nr Megablast.tsv", delim = "\t")

d <- data.frame(label = tree$tip.label)
d2 <- d %>%
  rename(Name = label) %>%
  left_join(nams, by = "Name") %>%
  mutate(Label = Description) %>%
  mutate(Label = gsub(" ","_",Label)) %>%
  mutate(Label = gsub('^([^_]+_[^_]+).*', '\\1', Label)) %>%
  mutate(Label = case_when(Label == "Fusarium_verticillioides" ~ "FFC", 
                           Label == "Fusarium_oxysporum" ~ "FOSC",
                           Label == "Fusarium_fujikuroi" ~ "FFC",
                           Label == "Fusarium_mangiferae" ~ "FFC",
                           Label == "Fusarium_proliferatum" ~ "FFC",
                           Label == "Fusarium_xylarioides" ~ "Arabica",
                           Label == "Fusarium_odoratissimum" ~ "FOSC",
                           grepl("Fusarium redolens", Description) ~ "Outgroup",
                           grepl("Foxy", Name) ~ "FOSC",
                           grepl("CP128283", Name) ~ "FOSC",
                           T ~ "Outgroup"))
pal <- c("#9C964A", "#F98400","lightgrey")

tree %>%
  ggtree() %<+% d2 +
  geom_tiplab(offset = 0.01, size = 3) +
  geom_nodelab(hjust = 1.5, size = 3) +
  geom_tippoint(aes(x = x+.01, color=Label), alpha=0.6,size=5, na.rm = TRUE) +
  labs(title = "HTR 3") +
  scale_color_manual(values = pal) +
  xlim(0,1) +
  theme(legend.position = "bottom",
        legend.title = element_blank()) +
  geom_treescale(y = -3)
#ggsave("htr3_tree.svg")

## then coverage plot
## closest tree tip to Fxyl is the same F. oxysporum as for HTR 2
foxy <- read_delim("./Data/GCA_0303451151_ASM3034511v1_genomic.minimap2_Fxyl_389563.LR.cov.binned.tsv", col_names = F)
foxy <- foxy %>%
  mutate(X1 = gsub("contig_","",X1)) %>%
  unite(X1,X2,X3,col = "key", sep = "-") %>%
  rename(GCA_030345115 = X4)
plot.tab <- all.new %>%
  unite(Chr:End,col = "key", sep = "-") %>%
  left_join(foxy, by = "key") %>%
  relocate(GCA_030345115, .before = Chr.new) %>%
  pivot_longer(cols = Robusta:GCA_030345115, names_to = "nams", values_to = "vals") %>%
  drop_na() %>%
  mutate(vals.new = log10(vals)) %>%
  mutate_if(is.numeric, ~ replace_na(., 0) %>% 
              replace(., is.infinite(.), -3))
pal <- c("#F5CDB4", "#F8AFA8","#F98400","#74A089")

plot.tab %>%
  ggplot(aes(x = Start.new, y = vals, col = nams)) +
  geom_line(alpha = 1) +
  coord_cartesian(xlim = c(37410000,37530000),ylim=c(0,1), clip = "off") +
  annotate("segment", x = 37440000, xend = 37480000, y = 1.1, yend = 1.1,colour = "blue") +
  xlim(37410000,37530000) +
  scale_color_manual(values = pal) +
  theme_bw() +
  scale_y_continuous(name = "Mapped coverage to arabica reference") +
  theme(legend.position = "bottom",
        legend.title = element_blank(),
        panel.grid.minor.x = element_blank(),
        panel.grid.major.y = element_blank(),
        panel.grid.minor.y = element_blank(),
        panel.grid.major.x = element_blank(),
        axis.title.y = element_blank(),
        axis.title.x = element_blank()) 

```


```{r HTR 4, Fig S6B}
## first tree
tree <- read.tree("./Data/htr4.treefile")
## table with strain details of BLAST hits
nams <- read_delim("./Data/104 documents from contig_11_460000-470000 - nr Megablast.tsv", delim = "\t")
tree$tip.label <- gsub("_1","",tree$tip.label)
d <- data.frame(label = tree$tip.label)
d2 <- d %>%
  rename(Name = label) %>%
  left_join(nams, by = "Name") %>%
  mutate(Label = Description) %>%
  mutate(Label = gsub(" ","_",Label)) %>%
  mutate(Label = gsub('^([^_]+_[^_]+).*', '\\1', Label)) %>%
  mutate(Label = case_when(Label == "Fusarium_verticillioides" ~ "FFC", 
                           Label == "Fusarium_oxysporum" ~ "FOSC",
                           Label == "Fusarium_fujikuroi" ~ "FFC",
                           Label == "Fusarium_mangiferae" ~ "FFC",
                           Label == "Fusarium_proliferatum" ~ "FFC",
                           Description == "Fusarium xylarioides 375908" ~ "Arabica",
                           Description == "Fusarium xylarioides 389563" ~ "Arabica",
                           Description == "Fusarium xylarioides 507038" ~ "Arabica",
                           grepl("Foxy", Name) ~ "FOSC",
                           T ~ "Outgroup"))
pal <- c("#9C964A", "#F98400","lightgrey","black")
tree %>%
  ggtree() %<+% d2 +
  geom_tiplab(offset = 0.03, size = 3) +
  geom_nodelab(hjust = 1.5, size = 3) +
  geom_tippoint(aes(x = x+.01, color=Label), alpha=0.6,size=5, na.rm = TRUE) +
  labs(title = "Contig 11") +
  scale_color_manual(values = pal) +
  xlim(0,1) +
  theme(legend.position = "bottom",
        legend.title = element_blank()) +
  geom_treescale(y = -1)

## then coverage plot
## closest tree tip to Fxyl is the F. oxysporum 509 which was sequenced in this study
foxy <- bedG %>%
  filter(Type == "Foxy509") %>%
  mutate(key = gsub("contig_","",key)) %>%
  rename(Foxy509 = Similarity) %>%
  dplyr::select(key:Foxy509)

plot.tab <- all.new %>%
  unite(Chr:End,col = "key", sep = "-") %>%
  left_join(foxy, by = "key") %>%
  relocate(Foxy509, .before = Chr.new) %>%
  pivot_longer(cols = Robusta:Foxy509, names_to = "nams", values_to = "vals") %>%
  drop_na() %>%
  mutate(vals.new = log10(vals)) %>%
  mutate_if(is.numeric, ~ replace_na(., 0) %>% 
              replace(., is.infinite(.), -3))

pal <- c("#F5CDB4", "#F8AFA8","#F98400","#74A089")
plot.tab %>%
  ggplot(aes(x = Start.new, y = vals, col = nams)) +
  geom_line(alpha = 1) +
  coord_cartesian(xlim = c(44990000,45090000),ylim=c(0,1), clip = "off") +
  annotate("segment", x = 45030000, xend = 45050000, y = 1.1, yend = 1.1,colour = "blue") +
  xlim(44990000,45090000) +
  scale_color_manual(values = pal) +
  theme_bw() +
  scale_y_continuous(name = "Mapped coverage to arabica reference") +
  theme(legend.position = "bottom",
        legend.title = element_blank(),
        panel.grid.minor.x = element_blank(),
        panel.grid.major.y = element_blank(),
        panel.grid.minor.y = element_blank(),
        panel.grid.major.x = element_blank(),
        axis.title.y = element_blank(),
        axis.title.x = element_blank()) 
```

```{r HTR 5, Fig S6C}
## first tree
tree <- read.tree("./Data/htr5.treefile")
## table with strain details of BLAST hits
nams <- read_delim("./Data/106 documents from contig_6_10000-250000 part 1 - nr Megablast.tsv", delim = "\t")
d <- data.frame(label = tree$tip.label)
d2 <- d %>%
  rename(Name = label) %>%
  left_join(nams, by = "Name") %>%
  mutate(Label = Description) %>%
  mutate(Label = gsub(" ","_",Label)) %>%
  mutate(Label = gsub('^([^_]+_[^_]+).*', '\\1', Label)) %>%
  mutate(Label = case_when(Label == "Fusarium_verticillioides" ~ "FFC", 
                           Label == "Fusarium_oxysporum" ~ "FOSC",
                           Label == "Fusarium_odoratissimum" ~ "FOSC",
                           grepl("Fxyl676", Name) ~ "Coffea1",
                           grepl("Fxyl659", Name) ~ "Coffea1",
                           grepl("Fxyl674", Name) ~ "Coffea2",
                           grepl("Fxyl113", Name) ~ "Coffea2",
                           grepl("Fxyl035", Name) ~ "Coffea2",
                           grepl("Fxyl038", Name) ~ "Arabica",
                           grepl("Fxyl563", Name) ~ "Arabica",
                           grepl("Fxyl908", Name) ~ "Arabica",
                           grepl("Fxyl254", Name) ~ "Robusta",
                           grepl("Fxyl277", Name) ~ "Robusta",
                           grepl("Fxyl394", Name) ~ "Robusta",
                           grepl("Fxyl268", Name) ~ "Robusta",
                           grepl("Fxyl925", Name) ~ "Robusta",
                           grepl("Foxy", Name) ~ "FOSC",
                           T ~ "Outgroup"))

pal <- c("#9C964A",  "#F98400")
tree %>%
  ggtree() %<+% d2 +
  geom_tiplab(offset = 0.1, size = 3) +
  geom_nodelab(hjust = 1.5, size = 3) +
  geom_tippoint(aes(x = x+.01, color=Label), alpha=0.6,size=5, na.rm = TRUE) +
  labs(title = "Contig 6") +
  scale_color_manual(values = pal) +
  xlim(0,5) +
  theme(legend.position = "bottom",
        legend.title = element_blank()) +
  geom_treescale(offset = -1)

### coverage plot
foxy <- read_delim("./Data/GCA_0303451151_ASM3034511v1_genomic.minimap2_Fxyl_389563.LR.cov.binned.tsv", col_names = F)
foxy <- foxy %>%
  mutate(X1 = gsub("contig_","",X1)) %>%
  unite(X1,X2,X3,col = "key", sep = "-") %>%
  rename(GCA_030345115 = X4)
plot.tab <- all.new %>%
  unite(Chr:End,col = "key", sep = "-") %>%
  left_join(foxy, by = "key") %>%
  relocate(GCA_030345115, .before = Chr.new) %>%
  pivot_longer(cols = Robusta:GCA_030345115, names_to = "nams", values_to = "vals") %>%
  drop_na() %>%
  mutate(vals.new = log10(vals)) %>%
  mutate_if(is.numeric, ~ replace_na(., 0) %>% 
              replace(., is.infinite(.), -3))
pal <- c("#F5CDB4", "#F8AFA8","#F98400","#74A089")

plot.tab %>%
  ggplot(aes(x = Start.new, y = vals, col = nams)) +
  geom_line(alpha = 1) +
  coord_cartesian(xlim = c(52480000,52620000),ylim=c(0,1), clip = "off") +
  annotate("segment", x = 52510000, xend = 52590000, y = 1.1, yend = 1.1,colour = "blue") +
  xlim(52480000,52620000) +
  scale_color_manual(values = pal) +
  theme_bw() +
  scale_y_continuous(name = "Mapped coverage to arabica reference") +
  theme(legend.position = "bottom",
        legend.title = element_blank(),
        panel.grid.minor.x = element_blank(),
        panel.grid.major.y = element_blank(),
        panel.grid.minor.y = element_blank(),
        panel.grid.major.x = element_blank(),
        axis.title.y = element_blank(),
        axis.title.x = element_blank()) 
```

```{r HTR 6, Fig S6D}
## first tree
tree <- read.tree("./Data/htr6.treefile")
## table with strain details of BLAST hits
nams <- read_delim("./Data/110 documents from scaffold_709 - nr Megablast.tsv", delim = "\t")
tree$tip.label <- gsub("_1","",tree$tip.label)
tree$tip.label <- gsub("_2","",tree$tip.label)
d <- data.frame(label = tree$tip.label)
d2 <- d %>%
  rename(Name = label) %>%
  left_join(nams, by = "Name") %>%
  mutate(Label = Description) %>%
  mutate(Label = gsub(" ","_",Label)) %>%
  mutate(Label = gsub('^([^_]+_[^_]+).*', '\\1', Label)) %>%
  mutate(Label = case_when(Label == "Fusarium_verticillioides" ~ "FFC", 
                           Label == "Fusarium_oxysporum" ~ "FOSC",
                           Label == "Fusarium_fujikuroi" ~ "FFC",
                           Label == "Fusarium_mangiferae" ~ "FFC",
                           Label == "Fusarium_proliferatum" ~ "FFC",
                           Description == "Fusarium xylarioides coffea1" ~ "Coffea1",
                           Description == "Fusarium xylarioides robusta" ~ "Robusta",
                           Description == "Fusarium xylarioides coffea2" ~ "Coffea2",
                           grepl("Foxy", Name) ~ "FOSC",
                           grepl("poae", Description) ~ "Outgroup",
                           T ~ "Outgroup"))
pal <- c("#F5CDB4", "#F8AFA8","#F98400","#74A089")

tree %>%
  ggtree() %<+% d2 +
  geom_tiplab(offset = 0.1, size = 3) +
  geom_nodelab(hjust = 1.5, size = 3) +
  geom_tippoint(aes(x = x+.01, color=Label), alpha=0.6,size=5, na.rm = TRUE) +
  labs(title = "Scaffold 709") +
  scale_color_manual(values = pal) +
  xlim(0,2.5) +
  theme(legend.position = "bottom",
        legend.title = element_blank()) +
  geom_treescale(y = -1)

## no coverage plot because HTR 6 is absent from the arabica population and therefore can't be mapped to the Fxyl reference genome
```